src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-snapshot-form/rbd-snapshot-form-modal.component.html

```diff
@@ -9,26 +9,43 @@
           novalidate>
       <div class="modal-body">
         <!-- Name -->
-        <div class="form-group row">
-          <label class="cd-col-form-label required"
-                 for="snapshotName"
-                 i18n>Name</label>
-          <div class="cd-col-form-input">
-            <input class="form-control"
-                   type="text"
-                   placeholder="Snapshot name..."
-                   id="snapshotName"
-                   name="snapshotName"
-                   [attr.disabled]="(mirroring === 'snapshot') ? true : null"
-                   formControlName="snapshotName"
-                   autofocus>
-            <span class="invalid-feedback"
-                  *ngIf="snapshotForm.showError('snapshotName', formDir, 'required')"
-                  i18n>This field is required.</span><br><br>
-            <span *ngIf="mirroring === 'snapshot'"
-                  i18n>Snapshot mode is enabled on image <b>{{ imageName }}</b>: snapshot names are auto generated</span>
+          <div class="form-group row">
+            <label class="cd-col-form-label required"
+                   for="snapshotName"
+                   i18n>Name</label>
+            <div class="cd-col-form-input">
+              <input class="form-control"
+                     type="text"
+                     placeholder="Snapshot name..."
+                     id="snapshotName"
+                     name="snapshotName"
+                     [attr.disabled]="((mirroring === 'snapshot') ? true : null) && (snapshotForm.getValue('mirrorImageSnapshot') === true) ? true: null"
+                     formControlName="snapshotName"
+                     autofocus>
+              <span class="invalid-feedback"
+                    *ngIf="snapshotForm.showError('snapshotName', formDir, 'required')"
+                    i18n>This field is required.</span><br><br>
+              <span *ngIf="((mirroring === 'snapshot') ? true : null) && (snapshotForm.getValue('mirrorImageSnapshot') === true) ? true: null"
+                    i18n>Snapshot mode is enabled on image <b>{{ imageName }}</b>: snapshot names are auto generated</span>
+            </div>
+          </div>
+          <div *ngIf="(mirroring === 'snapshot') ? true : null">
+            <div class="form-group row">
+              <div class="cd-col-form-offset">
+                <div class="custom-control custom-checkbox">
+                  <input type="checkbox"
+                         class="custom-control-input"
+                         formControlName="mirrorImageSnapshot"
+                         name="mirrorImageSnapshot"
+                         id="mirrorImageSnapshot"
+                         (change)="onMirrorCheckBoxChange()">
+                  <label for="mirrorImageSnapshot"
+                         class="custom-control-label"
+                         i18n>Mirror Image Snapshot</label>
+                </div>
+              </div>
```

Comment:
I prefer the checkbox to be above the name since this checkbox affects the `Name` input box.